src/hotspot/share/opto/subnode.cpp

```diff
@@ -1529,10 +1529,46 @@ Node *BoolNode::Ideal(PhaseGVN *phase, bool can_reshape) {
     }
   }
 
+  const int cmp1_op = cmp1->Opcode();
+  const int cmp2_op = cmp2->Opcode();
+
+  // Change x +- Integer.MIN_VALUE <=> y +- Integer.MIN_VALUE into x u<=> y
+  if ((_test._test == BoolTest::lt || _test._test == BoolTest::le ||
+        _test._test == BoolTest::gt || _test._test == BoolTest::ge) &&
+      cop == Op_CmpI &&
+      (cmp1_op == Op_AddI || cmp1_op == Op_SubI) &&
+      phase->type(cmp1->in(2)) == TypeInt::MIN) {
+    if (cmp2_op == Op_ConI) {
+      Node *ncmp2 = phase->intcon(java_add(cmp2->get_int(), min_jint));
```

Comment:
What if `cmp1_op == Op_SubI` ?

Reply:
Since `x + MIN_VALUE == x - MIN_VALUE`, the operation is the same with addition and subtraction operations. However, I realised that `x - MIN_VALUE` is idealised into `x + MIN_VALUE` beforehand, so I have removed the check for `Op_SubI` here.